client/src/main/java/io/pravega/client/tables/KeyValueTable.java

```diff
@@ -98,197 +122,147 @@
-     * <ul>
-     * <li>{@link ConditionalTableUpdateException} If the Key is not present present in the {@link KeyValueTable} for the
-     * provided Key Family or it is and has a different {@link Version}.
-     * See the {@link KeyValueTable} doc for more details on Conditional Update Responses.
-     * </ul>
-     */
-    CompletableFuture<Version> replace(@Nullable String keyFamily, @NonNull KeyT key, @NonNull ValueT value,
-                                       @NonNull Version version);
-
-    /**
-     * Inserts new or updates existing {@link TableEntry} instances that belong to the same Key Family into this
-     * {@link KeyValueTable}. All changes are performed atomically (either all or none will be accepted).
-     *
-     * @param keyFamily The Key Family for the all provided {@link TableEntry} instances.
-     * @param entries   An {@link Iterable} of {@link TableEntry} instances to insert or update. If for at least one
-     *                  such entry, {@link TableEntry#getKey()}{@link TableKey#getVersion()} indicates a conditional
-     *                  update, this will perform an atomic Conditional Update conditioned on the server-side versions
-     *                  matching the provided ones (for all {@link TableEntry} instances that have one); otherwise an
-     *                  Unconditional Update will be performed.
-     *                  See {@link KeyValueTable} doc for more details on Types of Updates.
-     * @return A CompletableFuture that, when completed, will contain a List of {@link Version} instances which
-     * represent the versions for the inserted/updated keys. The size of this list will be the same as entries.size()
-     * and the versions will be in the same order as the entries. Notable exceptions:
-     * <ul>
-     * <li>{@link ConditionalTableUpdateException} If this is a Conditional Update and the condition was not satisfied.
-     * See the {@link KeyValueTable} doc for more details on Conditional Update Responses.
-     * </ul>
-     */
-    CompletableFuture<List<Version>> replaceAll(@NonNull String keyFamily, @NonNull Iterable<TableEntry<KeyT, ValueT>> entries);
-
-    /**
-     * Unconditionally removes a Key from this Table Segment. If the Key does not exist, no action will be taken.
+     * Removes a {@link TableKey} from the {@link KeyValueTable}.
+     * <p>
+     * If {@link TableKey#isVersioned()} returns true, this will be a Conditional Removal. The {@link TableKey} will only
+     * be removed if there exists a {@link TableKey} in the {@link KeyValueTable} matching the given {@code key} and {@link Version}.
+     * <p>
+     * Otherwise, it will be executed as an Unconditional Removal ({@link TableKey#getVersion()} returns
+     * {@link Version#NO_VERSION} - such as when the key was created using {@link TableKey#unversioned}).
+     * <p>
+     * NOTE: it is possible to invoke this method with a {@link TableKey} created using {@link TableKey#notExists}. This
```

Comment:
We have an `exists` call below, so detecting whether a key exists should really be done via that call. I'm wondering if we should simply not allow `notExists` with remove.

Reply:
We could. It will have no effect though. In my opinion, if the operation is harmless, then documenting it as such (so users don't get confused about it) should be sufficient.